shell/platform/android/io/flutter/embedding/FlutterFragment.java

```diff
@@ -0,0 +1,425 @@
+    // background to a blank one since the launch screen is now in a view in front of the
+    // FlutterView.
+    //
+    // We can make this configurable if users want it.
+    // TODO: get rid of Activity references.
+    getActivity().setTheme(android.R.style.Theme_Black_NoTitleBar);
+
+    return view;
+  }
+
+  private boolean isSplashScreenDesired() {
+    return getArguments().getBoolean(ARG_IS_SPLASH_SCREEN_DESIRED, false);
+  }
+
+  /**
+   * Extracts a {@link Drawable} from the parent activity's {@code windowBackground}.
+   *
+   * {@code android:windowBackground} is specifically reused instead of other attributes
+   * because the Android framework can display it fast enough when launching the app as opposed
+   * to anything defined in the Activity subclass.
+   *
+   * TODO(mattcarroll): speed aside, should developers be given the opportunity to use different Drawables?
+   *
+   * Returns null if no {@code windowBackground} is set for the activity.
+   */
+  @SuppressWarnings("deprecation")
+  @Nullable
+  private Drawable getSplashScreenDrawableFromActivityTheme() {
+    TypedValue typedValue = new TypedValue();
+    if (!getContextCompat().getTheme().resolveAttribute(
+        android.R.attr.windowBackground,
+        typedValue,
+        true)) {
+      return null;
+    }
+    if (typedValue.resourceId == 0) {
+      return null;
+    }
+    try {
+      return getResources().getDrawable(typedValue.resourceId);
+    } catch (Resources.NotFoundException e) {
+      Log.e(TAG, "Referenced launch screen windowBackground resource does not exist");
+      return null;
+    }
+  }
+
+  /**
+   * Returns the Flutter view used by this {@code Fragment}; will be null before
+   * {@link #onCreateView(LayoutInflater, ViewGroup, Bundle)} is invoked. Will be
+   * non-null after {@link #onCreateView(LayoutInflater, ViewGroup, Bundle)} is invoked, up
+   * until {@link #onDestroyView()} is invoked.
+   */
+  @SuppressWarnings("unused")
+  @Nullable
+  public FlutterView getFlutterView() {
+    return flutterView;
+  }
+
+  /**
+   * Starts running Dart within the FlutterView for the first time.
+   *
+   * The difference between the first run and subsequent reloads is that when FlutterView is
+   * run for the first time, it attempts to reuse an existing Dart isolate.
+   */
+  private void doInitialFlutterViewRun() {
+    runFlutterView(getInitialRoute(), getAppBundlePath(), true);
+  }
+
+  /**
+   * Reloads Dart code with that at the given {@code appBundlePath}, sets Flutter's initial route
+   * to {@code initialRoute}, and then reloads the Dart VM.
+   *
+   * TODO(mattcarroll): is the above description accurate?
+   *
+   * @param initialRoute initial Flutter route to display after reloading Dart code
+   * @param appBundlePath Android file path to Flutter's app bundle
+   */
+  public void hotReload(@Nullable String initialRoute, @NonNull String appBundlePath) {
```

Comment:
Is this really hot reload? My understanding is that hot reload does not actually go through engine binding code. It is strictly related to Dart VM and its service channels.

Reply:
I honestly don't know.  I drew some assumptions based on what it looked like the original code was doing.  Feel free to provide any specific corrections.  It invokes the runFromBundle so I assumed it would reload and restart all Dart code...

Reply:
`FlutterView.runFromBundle` is used to start the engine.

Hot reload of new code into a running engine is done over the service protocol and does not invoke Java code.

Reply:
@jason-simmons if that's the case, then can you explain the execution path that involves an incoming ACTION_RUN Intent?

Reply:
ACTION_RUN is used by the `flutter run` command to launch the app during development.  `flutter run` will `adb shell` into the device and issue an intent that starts the app and provides engine configuration flags.

I don't think we currently have a scenario where ACTION_RUN is sent by the tool to an app process that is already running.  `flutter run` will install a new APK and kill any existing app process before restarting it with the intent.

(A long time ago we had a predecessor to hot reload that used an intent to signal that new Dart code should be loaded.  But that's obsolete now).

Reply:
If ACTION_RUN only has an effect during initial Flutter setup, do you know why this is implemented on onNewIntent()?  Wouldn't this behavior make sense in onCreate() where we can process the initial Intent one, and only one time?

Reply:
As far as I know, the use of onNewIntent is obsolete and is left over from the old implementation of code reload.  onCreate should be suitable for the current usage of the intent.